tests/scripts/run-corefx-tests.py

```diff
@@ -0,0 +1,297 @@
+    m = re.search('^\s*RID:\s+([^-]*)-(\S*)\s*$', dotnet_info, re.MULTILINE)
+    return m.group(1)
+
+
+def log(message):
+    """ Print logging information
+    Args:
+        message (str): message to be printed
+    """
+
+    print '[%s]: %s' % (sys.argv[0], message)
+
+
+##########################################################################
+# Main
+##########################################################################
+
+def main(args):
+    global Corefx_url
+    global Unix_name_map
+
+    arch, build_type, clr_root, fx_root, fx_branch, env_script = validate_args(
+        args)
+
+    clr_os = 'Windows_NT' if Is_windows else Unix_name_map[os.uname()[0]]
+
+    core_root = os.path.join(clr_root,
+                             'bin',
+                             'Product',
+                             '%s.%s.%s' % (clr_os, arch, build_type))
+
+    # corefx creates both files that are read-only and files that include non-ascii
+    # characters. Using onerror=del_rw allows us to delete all of the read-only files.
+    # To delete the files with non-ascii characters, when rmtree fails due to those
+    # files, we then will call rd on Windows.
+
+    if os.path.exists(fx_root):
+        if Is_windows:
+            vbcscompiler_running = True
+            while vbcscompiler_running:
+                res = subprocess.check_output(['tasklist'])
+                if not 'VBCSCompiler.exe' in res:
+                    vbcscompiler_running = False
+        os.chdir(fx_root)
+        os.system('git clean -fxd')
+        os.chdir(clr_root)
+        shutil.rmtree(fx_root, onerror=del_rw)
+
+    command = 'git clone -b %s --single-branch %s %s' % (
+        fx_branch, Corefx_url, fx_root)
+
+    log(command)
+
+    testing = False
+
+    if testing:
+        os.makedirs(fx_root)
+        returncode = 0
+    else:
+        returncode = os.system(command)
+
+    if returncode != 0:
+        sys.exit(returncode)
+
+    cwd = os.getcwd()
+    log('cd ' + fx_root)
+    os.chdir(fx_root)
+
+    if Is_windows:
+        command = '.\\build.cmd'
+        if env_script is not None:
+            command = ('cmd /c %s&&' % env_script) + command
+    else:
+        # CoreFx build.sh requires HOME to be set, and it isn't by default
+        # under our CI.
+        fx_home = os.path.join(fx_root, 'tempHome')
+        if not os.path.exists(fx_home):
+            os.makedirs(fx_home)
+        os.putenv('HOME', fx_home)
+        log('HOME=' + fx_home)
+
+        command = './build.sh'
+        if env_script is not None:
+            command = ('. %s;' % env_script) + command
+
+    if testing:
+        rid_os = dotnet_rid_os('')
+    else:
+        if clr_os == "Windows_NT":
+            rid_os = "win7"
+        else:
+            rid_os = dotnet_rid_os(os.path.join(clr_root, 'Tools', 'dotnetcli'))
+
+    command = ' '.join((
+        command,
+        '-Release',
+        '-TestNugetRuntimeId=%s-%s' % (rid_os, arch),
```

Comment:
For what it's worth, lifting the scripting out into a separate python script enables us to more easily test the script both locally before opening a PR, and also during testing, since we won't have regenerate the jobs, and we can run whatever job that we are trying to update (so we can make sure our scripting changes work on PR, not post-PR).